samples/bootloader/src/main.c

```diff
@@ -189,6 +189,23 @@ void main(void)
 	}
 #endif /* CONFIG_SOC_NRF9160 */
 
-	boot_from((u32_t *)s0_address_read());
+	const struct fw_firmware_info *s0_fw_info = fw_firmware_info_get(s0_address_read());
+	const struct fw_firmware_info *s1_fw_info = fw_firmware_info_get(s1_address_read());
+
+	if (s0_fw_info != NULL && s1_fw_info != NULL) {
```

Comment:
I've tried to find a way to simplify it but today I could not. I'll see if there is a better way of writing this tomorrow.

Reply:
looks good to me, legible enough.